servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/BlockingProcessorBuffer.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.api;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import javax.annotation.Nullable;
+
+import static io.servicetalk.concurrent.BlockingIterable.Processor;
+
+/**
+ * A buffer to store items for a {@link Processor}.
+ *
+ * @param <T>  Type of items stored in this buffer.
+ */
+public interface BlockingProcessorBuffer<T> {
+
+    /**
+     * Adds an item to this buffer.
+     *
+     * @param item to add.
+     * @throws InterruptedException If the add was interrupted.
+     */
+    void add(@Nullable T item) throws InterruptedException;
+
+    /**
+     * Terminates this buffer, such that no further modifications of this buffer are allowed. Subsequent
```

Comment:
maybe clarify what will happen if you try to call `add(...)` after it (maybe throwing).

Reply:
actually we are not enforcing this now, so lemme change `allowed` => `expected`